internal/generator/vector/output/otlp/otlp.go

```diff
@@ -0,0 +1,129 @@
+)
+
+func New(id string, o obsv1.OutputSpec, inputs []string, secrets vectorhelpers.Secrets, strategy common.ConfigStrategy, op Options) []Element {
+	if genhelper.IsDebugOutput(op) {
+		return []Element{
+			elements.Debug(helpers.MakeID(id, "debug"), vectorhelpers.MakeInputs(inputs...)),
+		}
+	}
+	// TODO: create a pattern to filter by input so all this is not necessary
+	var els []Element
+	// Creates reroutes for 'container','node','auditd','kubeAPI','openshiftAPI','ovn'
+	rerouteID := vectorhelpers.MakeID(id, "reroute")
+	els = append(els, RouteBySource(rerouteID, inputs))
+	// Container
+	transformContainerID := vectorhelpers.MakeID(id, logSourceContainer)
+	reduceContainerID := vectorhelpers.MakeID(id, "groupby", "container")
+	els = append(els, TransformContainer(transformContainerID, []string{rerouteID + "." + logSourceContainer}))
+	// Group by cluster_id, namespace_name, pod_name, container_name
+	els = append(els, GroupByContainer(reduceContainerID, []string{transformContainerID}))
+	// Journal
+	transformNodeID := vectorhelpers.MakeID(id, logSourceNode)
+	els = append(els, TransformJournal(transformNodeID, []string{rerouteID + "." + logSourceNode}))
+	// Audit
+	transformAuditHostID := vectorhelpers.MakeID(id, logSourceAuditd)
+	transformAuditKubeID := vectorhelpers.MakeID(id, logSourceKubeAPI)
+	transformAuditOpenshiftID := vectorhelpers.MakeID(id, logSourceOpenshiftAPI)
+	transformAuditOvnID := vectorhelpers.MakeID(id, logSourceOvn)
+	reduceSourceID := vectorhelpers.MakeID(id, "groupby", "source")
+	reduceHostID := vectorhelpers.MakeID(id, "groupby", "host")
+	els = append(els, TransformAuditHost(transformAuditHostID, []string{rerouteID + "." + logSourceAuditd}))
+	els = append(els, TransformAuditKube(transformAuditKubeID, []string{rerouteID + "." + logSourceKubeAPI}))
+	els = append(els, TransformAuditOpenshift(transformAuditOpenshiftID, []string{rerouteID + "." + logSourceOpenshiftAPI}))
+	els = append(els, TransformAuditOvn(transformAuditOvnID, []string{rerouteID + "." + logSourceOvn}))
+	// Group by cluster_id, log_source
+	els = append(els, GroupBySource(reduceSourceID, []string{
+		transformNodeID,
+		transformAuditKubeID,
+		transformAuditOpenshiftID,
+		transformAuditOvnID,
+	}))
+	// Group by cluster_id, hostname
+	els = append(els, GroupByHost(reduceHostID, []string{
+		transformAuditHostID,
+	}))
+
+	// Normalize all into resource and scopeLogs objects
+	formatResourceLogsID := vectorhelpers.MakeID(id, "resource", "logs")
+	els = append(els, FormatResourceLog(formatResourceLogsID, []string{
+		reduceContainerID,
+		reduceSourceID,
+		reduceHostID,
+		rerouteID + "._unmatched", // mostly for debug, but could be necessary?
+	}))
+	// Create sink and wrap in `resourceLogs`
+	sink := Output(id, o, []string{formatResourceLogsID}, secrets, op)
+	if strategy != nil {
+		strategy.VisitSink(sink)
+	}
+	return MergeElements(
+		els,
+		[]Element{
+			sink,
+			common.NewAcknowledgments(id, strategy),
```

Comment:
I'll add it for consistency of course, thanks.      Note: that field will never be included since we set specific fields when we create a new object at the end of our remaps.